Framework/Algorithms/src/FilterEvents.cpp

```diff
@@ -1117,7 +1118,7 @@ void FilterEvents::createOutputWorkspacesSplitters() {
         add2output = false;
     }
 
-    std::shared_ptr<EventWorkspace> optws = create<EventWorkspace>(*m_eventWS);
+    std::shared_ptr<EventWorkspace> optws = prototype_ws->clone();
     // Clear Run without copying first.
     optws->setSharedRun(Kernel::make_cow<Run>());
```

Comment:
Can this be moved outside the loop and just done once on `prototype_ws`? Same for the other loop.

Reply:
I don't think so?  The code looks like its using copies of this workspace to load up with data and insert into a map of unique output workspaces

Reply:
Could you try OpenMP on the loop?  I am not sure how much time that you can save for cloning hundreds of event workspaces.

Reply:
@wdzhou could you point me to an example? I have not used that before.

Reply:
@wdzhou That sounds like some that can be tried as part of ED88 or at least a separate task on ED87?

Reply:
@rosswhitfield This is just one of the idea to improve performance.  It may not be necessary in this PR.  

@walshmm Like this. https://github.com/mantidproject/mantid/blob/main/Framework/Algorithms/src/FitPeaks.cpp#L853